src/MDMTransaction.m

```diff
@@ -30,17 +32,58 @@ - (instancetype)init {
 }
 
 - (void)addPlan:(NSObject<MDMPlan> *)plan toTarget:(id)target {
-  MDMTransactionLog *log = [MDMTransactionLog new];
-  log.plans = @[ [plan copy] ];
-  log.target = target;
+  [self commonAddPlan:plan toTarget:target withName:nil];
+}
+
+- (void)addPlan:(NSObject<MDMPlan> *)plan toTarget:(id)target named:(NSString *)name {
+  if (name.length && [self planForName:name withTarget:target]) {
```

Comment:
There's two places where this PR checks for `name.length > 0`. Could be worthwhile to extract this into a method on `MDMTransactionLog` if you think it'd be useful.